src/rust/rust-proto/src/graplinc/grapl/common/v1beta1/types.rs

```diff
@@ -0,0 +1,245 @@
+use crate::{
+    protobufs::graplinc::grapl::common::v1beta1::{
+        EdgeName as EdgeNameProto,
+        NodeType as NodeTypeProto,
+        PropertyName as PropertyNameProto,
+        Uid as UidProto,
+    },
+    serde_impl,
+    type_url,
+    SerDeError,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct PropertyName {
+    pub value: String,
+}
+
+impl std::fmt::Display for PropertyName {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.value)
+    }
+}
+
+impl PropertyName {
+    pub fn new_unchecked(value: String) -> Self {
+        // todo: debug assertion
+        Self { value }
+    }
+}
+
+impl TryFrom<&'static str> for PropertyName {
+    type Error = SerDeError;
+    fn try_from(raw: &'static str) -> Result<Self, Self::Error> {
+        if raw.is_empty() {
+            return Err(SerDeError::InvalidField {
+                field_name: "PropertyName",
+                assertion: "can not be empty".to_owned(),
+            });
+        }
+        if raw.len() > 32 {
+            return Err(SerDeError::InvalidField {
+                field_name: "PropertyName",
+                assertion: "can not be more than 32 characters".to_owned(),
+            });
+        }
+
+        Ok(Self {
+            value: raw.to_owned(),
+        })
+    }
+}
+
+impl TryFrom<String> for PropertyName {
+    type Error = &'static str;
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        // todo: Validate this thing
+        Ok(PropertyName { value })
+    }
+}
+
+impl TryFrom<PropertyNameProto> for PropertyName {
```

Comment:
Why is this `TryFrom` when it's infallible?

Reply:
Changed

Reply:
Let me add a comment. Discussed with Colin over DM he says there will be future validation here

Reply:
This will be fallible in the future when validation is added to ensure that PropertyName is constrained.

Reply:
FWIW the validation is documented in the proto file - that's why many of these are infallible but will absolutely be fallible in the very near future.